test-util/src/main/java/io/zeebe/test/util/testcontainers/ManagedVolume.java

```diff
@@ -0,0 +1,93 @@
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.CreateContainerCmd;
+import com.github.dockerjava.api.command.CreateVolumeCmd;
+import com.github.dockerjava.api.command.CreateVolumeResponse;
+import com.github.dockerjava.api.model.Bind;
+import com.github.dockerjava.api.model.HostConfig;
+import java.util.Objects;
+import org.testcontainers.DockerClientFactory;
+
+/**
+ * A simple wrapper to create Docker volumes which are managed by Testcontainers. The created object
+ * labels the volumes with {@link DockerClientFactory#DEFAULT_LABELS} so that the Ryuk container can
+ * reap the volumes should our JVM process crash.
+ */
+public final class ManagedVolume {
+
+  @SuppressWarnings("java:S1075")
+  private static final String DEFAULT_ZEEBE_DATA_PATH = "/usr/local/zeebe/data";
+
+  private final String name;
+
+  /** @param name the name of the volume */
+  public ManagedVolume(final String name) {
+    this.name = name;
+  }
+
+  /** @return the name of the volume */
+  public String getName() {
+    return name;
+  }
+
+  /**
+   * Returns the volume as a bind which can be used when creating new containers. For example:
+   * <code>
+   *   new GenericContainer("containerImage").withCreatCmdModifier(cmd -> cmd.withHostConfig(
+   *      cmd.getHostConfig().withBinds(managedVolume.asBind("/path/to/mount"))));
+   * </code>
+   *
+   * @param mountPath the path where to mount the volume in the container
+   * @return a bind which can be used when creating a container
+   */
+  public Bind asBind(final String mountPath) {
+    return Bind.parse(name + ":" + mountPath);
+  }
+
+  /**
+   * @return a pre-configured {@link Bind} which mounts this volume to the data folder of a Zeebe *
+   *     broker.
+   */
+  public Bind asZeebeBind() {
+    return asBind(DEFAULT_ZEEBE_DATA_PATH);
+  }
+
+  /**
+   * Convenience method which mounts the volume to a Zeebe broker's data folder.
+   *
+   * @param command the create command of the Zeebe broker container
+   */
+  public void attachVolumeToContainer(final CreateContainerCmd command) {
+    final HostConfig hostConfig =
+        Objects.requireNonNull(command.getHostConfig()).withBinds(asZeebeBind());
+    command.withHostConfig(hostConfig);
+  }
+
+  /**
+   * @return a new managed volume using the default Docker client (see {@link
+   *     DockerClientFactory#instance()})
+   */
+  public static ManagedVolume newVolume() {
+    return newVolume(DockerClientFactory.instance().client());
+  }
+
+  /**
+   * @param client the Docker client to use to create the volume
+   * @return a new managed volume using the given Docker client to create it
+   */
+  public static ManagedVolume newVolume(final DockerClient client) {
+    try (final CreateVolumeCmd command = client.createVolumeCmd()) {
+      final CreateVolumeResponse response =
+          command.withLabels(DockerClientFactory.DEFAULT_LABELS).exec();
+      return new ManagedVolume(response.getName());
```

Comment:
You said that volumes are "properly managed by Testcontainers by adding the right labels". Is this what you meant? I tried googling a bit on how it works but didn't find much

Reply:
Yes, this is undocumented unfortunately, but Sergei is the one who told me to use this. I imagine one downside is it's a little brittle if they ever change how Ryuk works, but I think that's a small thing. Of course it would be nice if Testcontainers already provided the feature, and maybe that's something I'll contribute at some point so we have an official, maintained API on how to do this.